src/Microsoft.CodeAnalysis.Analyzers/Core/MetaAnalyzers/DiagnosticAnalyzerAttributeAnalyzer.cs

```diff
@@ -54,7 +54,8 @@ public override void Initialize(AnalysisContext context)
         [SuppressMessage("AnalyzerPerformance", "RS1012:Start action has no registered actions.", Justification = "Method returns an analyzer that is registered by the caller.")]
         protected override DiagnosticAnalyzerSymbolAnalyzer GetDiagnosticAnalyzerSymbolAnalyzer(CompilationStartAnalysisContext compilationContext, INamedTypeSymbol diagnosticAnalyzer, INamedTypeSymbol diagnosticAnalyzerAttribute)
         {
-            var attributeUsageAttribute = WellKnownTypes.AttributeUsageAttribute(compilationContext.Compilation);
+            var wellKnownTypeProvider = WellKnownTypeProvider.GetOrCreate(compilationContext.Compilation);
+            var attributeUsageAttribute = wellKnownTypeProvider.AttributeUsageAttribute;
```

Comment:
Can we avoid requiring each analyzer to have to get an instance of the provider? We can define an extension method `GetOrCreateWellKnownType` in [CompilationExtensions.cs](https://github.com/dotnet/roslyn-analyzers/blob/master/src/Utilities/Compiler/Extensions/CompilationExtensions.cs) that internally uses the WellKnownProvider so the analyzer can write this as:

```csharp
compilationContext.Compilation.GetOrCreateWellKnownType(WellKnownTypeNames.AttributeUsageAttribute);
```

Analyzers that need to pass around the well known type provider can explicitly invoke `WellKnownTypeProvider.GetOrCreate` to get the instance.